proto/internal/temporal/server/api/matchingservice/v1/request_response.proto

```diff
@@ -318,3 +319,60 @@ message ReplicateTaskQueueUserDataRequest {
 message ReplicateTaskQueueUserDataResponse {
 }
 
+message DispatchNexusTaskRequest {
+    string namespace_id = 1;
+    temporal.api.taskqueue.v1.TaskQueue task_queue = 2;
+    // Nexus request extracted by the frontend and translated into Temporal API format.
+    temporal.api.nexus.v1.Request request = 3;
+    // Non-empty if this task was forwarded from a child partition.
+    string forwarded_source = 4;
+}
+
+message DispatchNexusTaskResponse {
+    oneof outcome {
+        // Set if the worker's handler failed the nexus task.
+        temporal.api.nexus.v1.HandlerError handler_error = 1;
+        // Set if the worker's handler responded successfully to the nexus task.
+        temporal.api.nexus.v1.Response response = 2;
+    }
+}
+
+message PollNexusTaskQueueRequest {
+    string namespace_id = 1;
+    // A unique ID generated by the frontend for this request.
+    string poller_id = 2;
+    // Original WorkflowService poll request as received by the frontend.
+    temporal.api.workflowservice.v1.PollNexusTaskQueueRequest request = 3;
+    // Non-empty if this task was forwarded from a child partition.
+    string forwarded_source = 4;
+}
+
+message PollNexusTaskQueueResponse {
+    // Response that should be delivered to the worker containing a request from DispatchNexusTaskRequest.
+    temporal.api.workflowservice.v1.PollNexusTaskQueueResponse response = 1;
+}
+
+message RespondNexusTaskCompletedRequest {
+    string namespace_id = 1;
+    temporal.api.taskqueue.v1.TaskQueue task_queue = 2;
+    // A unique ID for this task generated by the matching engine. Decoded from the incoming request's task token.
+    string task_id = 3;
+    // Original completion as received by the frontend.
+    temporal.api.workflowservice.v1.RespondNexusTaskCompletedRequest request = 4;
+}
+
+message RespondNexusTaskCompletedResponse {
+}
+
+message RespondNexusTaskFailedRequest {
```

Comment:
why use a oneof on `DispatchNexusTaskResponse` but separate messages here? should there just be a single `RespondNexusTaskRequest` with a oneof?

Reply:
Consistency with the rest of our APIs.

Note that `DispatchNexusTaskResponse` has to convey both success and failure to unblock the frontend RPC similarly to how `QueryWorkflowResponse` is modeled.